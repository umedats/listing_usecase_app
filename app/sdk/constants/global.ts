const APP_NAME = "SellingPartnerAPIListingSampleApplication";
const APP_VERSION = "1.0";
const APP_LANGUAGE = "TypeScript";
const APP_LANGUAGE_VERSION = "5.0";
/**
 * Header data to be passed to all the SP-API.
 */
export const HEADER_DATA = {
  "User-Agent": `${APP_NAME}/${APP_VERSION} (Language=${APP_LANGUAGE}/${APP_LANGUAGE_VERSION})`,
};
