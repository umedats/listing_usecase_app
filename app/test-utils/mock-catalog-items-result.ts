export const MOCK_CATALOG_ITEMS_API_RESPONSE: object = {
  numberOfResults: 1,
  items: [
    {
      asin: "B00004SPEC",
      productTypes: [
        {
          marketplaceId: "ATVPDKIKX0DER",
          productType: "KITCHEN",
        },
      ],
      summaries: [
        {
          marketplaceId: "ATVPDKIKX0DER",
          adultProduct: false,
          autographed: false,
          brand: "KRUPS",
          browseClassification: {
            displayName: "Countertop Blenders",
            classificationId: "289915",
          },
          itemClassification: "BASE_PRODUCT",
          itemName: "Krups 243-70 Power X Plus Combi Blender, DISCONTINUED",
          manufacturer: "KRUPS",
          memorabilia: false,
          modelNumber: "243-70",
          packageQuantity: 1,
          partNumber: "243-70",
          style: "Modern",
          tradeInEligible: false,
          websiteDisplayGroup: "kitchen_display_on_website",
          websiteDisplayGroupName: "Kitchen",
        },
      ],
    },
  ],
};

export const MOCK_CATALOG_ITEMS_API_EMPTY_RESPONSE: object = {
  numberOfResults: 0,
};
