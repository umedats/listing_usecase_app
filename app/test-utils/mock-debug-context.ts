import { SPAPIRequestResponse } from "@/app/model/types";

export const MOCK_SP_API_RESPONSE: SPAPIRequestResponse = {
  apiName: "GetListingsItem",
  apiDocumentationLink: "https://amazon.com",
  request: {
    marketplaceId: "ATVPDKIKX0DER",
    locale: "en_US",
    asin: "A1F83G8C2ARO7P",
  },
  response: {
    item: "Dummy Item",
  },
};
