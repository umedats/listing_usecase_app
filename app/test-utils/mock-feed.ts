export const MOCK_FEED_1 = {
  feedId: "1000",
  feedType: "JSON_LISTINGS_FEED",
  createdTime: "2023-10-05T04:23:58+0000",
  processingStatus: "IN_PROGRESS",
  resultFeedDocumentId:
    "amzn1.tortuga.3.ed4cd0d8-447b-4c22-96b5-52da8ace1207.T3YUVYPGKE9BMY",
};

export const MOCK_FEED_2 = {
  feedId: "2000",
  feedType: "JSON_LISTINGS_FEED",
  createdTime: "2023-10-05T04:23:58+0000",
  processingStatus: "IN_PROGRESS",
};

export const MOCK_CREATE_FEED_DOCUMENT_SUCCESS = {
  feedDocumentId: "123456",
  url: "https:feed.document/url",
};

export const MOCK_CREATE_FEED_SUCCESS = {
  feedId: "123456789",
};
