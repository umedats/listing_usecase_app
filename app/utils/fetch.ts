import { SPAPIRequestResponse } from "@/app/model/types";
import { SettableDebugState } from "@/app/context/debug-context-provider";
import { NO_CACHING_DIRECTIVE } from "@/app/constants/global";

/**
 * Helper function to fetch the data from the backend API. This function
 * extracts the debug context, and invokes the updateDebugContextCallback
 * method.
 * @param url the url of the backend API.
 * @param updateDebugContextCallback a callback to update the debug context.
 * @param responseDataCallback an optional call back which is invoked with the
 * data extracted from the backend API.
 * @param requestInit params to initialize the fetch request.
 */
export async function fetcherForBackendAPI(
  url: string,
  updateDebugContextCallback: (debugData: any) => void,
  responseDataCallback?: (data: any) => any | void,
  requestInit?: RequestInit,
) {
  let newRequestInit = requestInit;
  if (requestInit) {
    newRequestInit = { ...requestInit, cache: NO_CACHING_DIRECTIVE };
  }

  const response = await fetch(url, newRequestInit);
  const responseBody = await response.json();
  if (responseBody.debugContext) {
    updateDebugContextCallback(responseBody.debugContext);
  }

  if (!response.ok) {
    throw [response.status, response.statusText];
  }

  const data = responseBody.data;
  if (responseDataCallback) {
    return responseDataCallback(data);
  }
  return data;
}

/**
 * Saves the given debugData for the given route into debugContext.
 * @param pathname
 * @param debugData
 * @param debugContext
 */
export function updateDebugContext(
  pathname: string,
  debugData: any,
  debugContext: SettableDebugState,
) {
  (debugData as SPAPIRequestResponse[]).forEach((value) =>
    debugContext.addRequestResponse(pathname, value),
  );
}
