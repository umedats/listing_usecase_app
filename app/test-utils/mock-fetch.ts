import fetch from "jest-fetch-mock";

/**
 * Mocks a fetch call with the Promise.resolve().
 * @param status the HTTP status of the mocked response.
 * @param body any body of the mocked response.
 */
export function mockResolveFetchResponse(status: number, body: object) {
  fetch.mockResponse(() =>
    Promise.resolve({
      status: status,
      body: JSON.stringify(body),
    }),
  );
}

/**
 * Mocks a fetch call with the Promise.resolve(). The given body is not
 * serialized into Json.
 * @param status the HTTP status of the mocked response.
 * @param body any body of the mocked response.
 * @param headers the headers for the response.
 */
export function mockResolveFetchResponseNoJsonSerialization(
  status: number,
  body: any,
  headers?: any,
) {
  fetch.mockResponse(() =>
    Promise.resolve({
      status: status,
      body: body,
      headers: headers,
    }),
  );
}

/**
 * Mocks a fetch call with the Promise.reject().
 * @param status the HTTP status of the mocked response.
 */
export function mockRejectFetchResponse(status: number) {
  fetch.mockResponse(() =>
    Promise.reject({
      status: status,
    }),
  );
}

/**
 * Mocks a fetch response by introducing a delay.
 * @param status the HTTP response status
 * @param body the body of the response
 * @param delayInMillis the delay for the promise resolution.
 */
export function mockDelayedResolveFetchResponse(
  status: number,
  body: object,
  delayInMillis: number,
) {
  const delayedPromise = new Promise((resolve) => {
    setTimeout(
      () =>
        resolve({
          status: status,
          body: JSON.stringify(body),
        }),
      delayInMillis,
    );
  });
  // @ts-ignore
  fetch.mockResponse(() => delayedPromise);
}
