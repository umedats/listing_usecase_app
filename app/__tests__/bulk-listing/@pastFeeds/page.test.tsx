import "@testing-library/jest-dom";
import { act, render, screen, waitFor } from "@testing-library/react";
import React from "react";
import { SWRConfig } from "swr";
import DebuggingContextProvider from "@/app/context/debug-context-provider";
import { NextIntlProvider } from "next-intl";
import PastFeeds from "@/app/[locale]/bulk-listing/@pastFeeds/page";
import translations from "@/app/internationalization/translations/en-US.json";
import fetch from "jest-fetch-mock";
import { mockResolveFetchResponse } from "@/app/test-utils/mock-fetch";
import { MOCK_FEED_1, MOCK_FEED_2 } from "@/app/test-utils/mock-feed";
import { MOCK_SP_API_RESPONSE } from "@/app/test-utils/mock-debug-context";
import userEvent from "@testing-library/user-event";
import { US_LOCALE } from "@/app/constants/global";

function renderPastFeeds() {
  return render(
    <SWRConfig value={{ provider: () => new Map() }}>
      <DebuggingContextProvider>
        <NextIntlProvider locale={US_LOCALE} messages={translations}>
          <PastFeeds />
        </NextIntlProvider>
      </DebuggingContextProvider>
    </SWRConfig>,
  );
}
describe("Test for the Past Feeds page", () => {
  beforeEach(() => {
    fetch.resetMocks();
  });

  test("snapshot test which verifies if the feed data is rendered properly", async () => {
    mockResolveFetchResponse(200, {
      data: [MOCK_FEED_1, MOCK_FEED_2],
      debugContext: [MOCK_SP_API_RESPONSE],
    });

    await renderAndValidate();
  });

  test("snapshot test which verifies if the alert is displayed for no feeds", async () => {
    mockResolveFetchResponse(200, {
      data: [],
      debugContext: [MOCK_SP_API_RESPONSE],
    });

    await renderAndValidate();
  });
});

async function renderAndValidate() {
  const { asFragment } = renderPastFeeds();
  expect(asFragment()).toMatchSnapshot();

  await waitFor(() =>
    expect(screen.queryByTestId("getFeedsButton")).not.toBeNull(),
  );
  await act(async () => {
    await userEvent.click(screen.getByTestId("getFeedsButton"));
  });
  expect(asFragment()).toMatchSnapshot();
}
