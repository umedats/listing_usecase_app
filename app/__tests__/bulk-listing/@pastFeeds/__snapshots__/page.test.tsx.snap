// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the Past Feeds page snapshot test which verifies if the alert is displayed for no feeds 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthMd css-xdmu94-MuiContainer-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-13mqtjd-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Past Submitted Json Listing Feeds
        </h6>
      </div>
    </div>
    <span
      aria-label="Click the button to retrieve the details of past submitted Json Listing Feeds"
      class=""
      data-mui-internal-clone-element="true"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
        data-testid="getFeedsButton"
        id="getFeedsButton"
        tabindex="0"
        type="button"
      >
        View Feeds
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Test for the Past Feeds page snapshot test which verifies if the alert is displayed for no feeds 2`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthMd css-xdmu94-MuiContainer-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-13mqtjd-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Past Submitted Json Listing Feeds
        </h6>
      </div>
    </div>
    <span
      aria-label="Click the button to retrieve the details of past submitted Json Listing Feeds"
      class=""
      data-mui-internal-clone-element="true"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
        data-testid="getFeedsButton"
        id="getFeedsButton"
        tabindex="0"
        type="button"
      >
        View Feeds
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </span>
    <div
      class="MuiBox-root css-y1gt6f"
    >
      <div
        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardInfo MuiAlert-standard css-ethv1g-MuiPaper-root-MuiAlert-root"
              data-testid="no-feeds-alert"
              role="alert"
            >
              <div
                class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="InfoOutlinedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
                  />
                </svg>
              </div>
              <div
                class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
              >
                No past feeds found.
              </div>
              <div
                class="MuiAlert-action css-ki1hdl-MuiAlert-action"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Test for the Past Feeds page snapshot test which verifies if the feed data is rendered properly 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthMd css-xdmu94-MuiContainer-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-13mqtjd-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Past Submitted Json Listing Feeds
        </h6>
      </div>
    </div>
    <span
      aria-label="Click the button to retrieve the details of past submitted Json Listing Feeds"
      class=""
      data-mui-internal-clone-element="true"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
        data-testid="getFeedsButton"
        id="getFeedsButton"
        tabindex="0"
        type="button"
      >
        View Feeds
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Test for the Past Feeds page snapshot test which verifies if the feed data is rendered properly 2`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthMd css-xdmu94-MuiContainer-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-13mqtjd-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Past Submitted Json Listing Feeds
        </h6>
      </div>
    </div>
    <span
      aria-label="Click the button to retrieve the details of past submitted Json Listing Feeds"
      class=""
      data-mui-internal-clone-element="true"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
        data-testid="getFeedsButton"
        id="getFeedsButton"
        tabindex="0"
        type="button"
      >
        Refresh Feeds
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </span>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-3su884-MuiPaper-root"
      style="padding: 10px; margin: 15px;"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-18ksmt-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            >
              <label
                aria-label="The identifier for the feed. This identifier is unique only in combination with a seller ID"
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
                data-mui-internal-clone-element="true"
                for="FeedContainer-0-id"
              >
                FeedID
              </label>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
            >
              <div
                aria-label="The identifier for the feed. This identifier is unique only in combination with a seller ID"
                class="MuiBox-root css-1v3tnle"
                data-mui-internal-clone-element="true"
                id="FeedContainer-0-id"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  1000
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-18ksmt-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            >
              <label
                aria-label="Feed Type of the submission"
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
                data-mui-internal-clone-element="true"
                for="FeedContainer-0-type"
              >
                FeedType
              </label>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
            >
              <div
                aria-label="Feed Type of the submission"
                class="MuiBox-root css-1v3tnle"
                data-mui-internal-clone-element="true"
                id="FeedContainer-0-type"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  JSON_LISTINGS_FEED
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-18ksmt-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            >
              <label
                aria-label="The date and time when the feed was created, in ISO 8601 date time format"
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
                data-mui-internal-clone-element="true"
                for="FeedContainer-0-createdTime"
              >
                Created Time
              </label>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
            >
              <div
                aria-label="The date and time when the feed was created, in ISO 8601 date time format"
                class="MuiBox-root css-1v3tnle"
                data-mui-internal-clone-element="true"
                id="FeedContainer-0-createdTime"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  2023-10-05T04:23:58+0000
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-18ksmt-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            >
              <label
                aria-label="The processing status of the feed submission"
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
                data-mui-internal-clone-element="true"
                for="FeedContainer-0-processingStatus"
              >
                Processing Status
              </label>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
            >
              <div
                aria-label="The processing status of the feed submission"
                class="MuiBox-root css-1v3tnle"
                data-mui-internal-clone-element="true"
                id="FeedContainer-0-processingStatus"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  IN_PROGRESS
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <span
            aria-label="Click the button to download the processing report"
            class=""
            data-mui-internal-clone-element="true"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
              data-testid="downloadProcessingReport-0"
              id="downloadProcessingReport-0"
              tabindex="0"
              type="button"
            >
              Download Processing Report
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </span>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-3su884-MuiPaper-root"
      style="padding: 10px; margin: 15px;"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-18ksmt-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            >
              <label
                aria-label="The identifier for the feed. This identifier is unique only in combination with a seller ID"
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
                data-mui-internal-clone-element="true"
                for="FeedContainer-1-id"
              >
                FeedID
              </label>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
            >
              <div
                aria-label="The identifier for the feed. This identifier is unique only in combination with a seller ID"
                class="MuiBox-root css-1v3tnle"
                data-mui-internal-clone-element="true"
                id="FeedContainer-1-id"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  2000
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-18ksmt-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            >
              <label
                aria-label="Feed Type of the submission"
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
                data-mui-internal-clone-element="true"
                for="FeedContainer-1-type"
              >
                FeedType
              </label>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
            >
              <div
                aria-label="Feed Type of the submission"
                class="MuiBox-root css-1v3tnle"
                data-mui-internal-clone-element="true"
                id="FeedContainer-1-type"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  JSON_LISTINGS_FEED
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-18ksmt-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            >
              <label
                aria-label="The date and time when the feed was created, in ISO 8601 date time format"
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
                data-mui-internal-clone-element="true"
                for="FeedContainer-1-createdTime"
              >
                Created Time
              </label>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
            >
              <div
                aria-label="The date and time when the feed was created, in ISO 8601 date time format"
                class="MuiBox-root css-1v3tnle"
                data-mui-internal-clone-element="true"
                id="FeedContainer-1-createdTime"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  2023-10-05T04:23:58+0000
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-18ksmt-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            >
              <label
                aria-label="The processing status of the feed submission"
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
                data-mui-internal-clone-element="true"
                for="FeedContainer-1-processingStatus"
              >
                Processing Status
              </label>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
            >
              <div
                aria-label="The processing status of the feed submission"
                class="MuiBox-root css-1v3tnle"
                data-mui-internal-clone-element="true"
                id="FeedContainer-1-processingStatus"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  IN_PROGRESS
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <span
            aria-label="Click the button to download the processing report"
            class=""
            data-mui-internal-clone-element="true"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
              data-testid="downloadProcessingReport-1"
              disabled=""
              id="downloadProcessingReport-1"
              tabindex="-1"
              type="button"
            >
              Download Processing Report
            </button>
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
